packages/editor/src/browser/editor-linenumber-contribution.ts

```diff
@@ -0,0 +1,86 @@
+// *****************************************************************************
+// Copyright (C) 2023 STMicroelectronics and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { EditorManager } from './editor-manager';
+import { EditorMouseEvent, MouseTargetType, Position, TextEditor } from './editor';
+import { injectable, inject } from '@theia/core/shared/inversify';
+import { FrontendApplicationContribution, ContextMenuRenderer } from '@theia/core/lib/browser';
+import { ContextKeyService } from '@theia/core/lib/browser/context-key-service';
+import { DisposableCollection, MenuPath } from '@theia/core';
+import { EditorWidget } from './editor-widget';
+
+export const EDITOR_LINENUMBER_CONTEXT_MENU: MenuPath = ['editor_linenumber_context_menu'];
+
+@injectable()
+export class EditorLineNumberContribution implements FrontendApplicationContribution {
+
+    @inject(ContextKeyService)
+    protected readonly contextKeyService: ContextKeyService;
+
+    @inject(ContextMenuRenderer)
+    protected readonly contextMenuRenderer: ContextMenuRenderer;
+
+    @inject(EditorManager)
+    protected readonly editorManager: EditorManager;
+
+    onStart(): void {
+        this.editorManager.onCreated(editor => this.addLineNumberContextMenu(editor));
+    }
+
+    protected addLineNumberContextMenu(editorWidget: EditorWidget): void {
+        const editor = editorWidget.editor;
+        if (editor) {
+            const disposables = new DisposableCollection();
+            disposables.push(editor.onMouseDown(event => this.handleContextMenu(editor, event)));
+            editorWidget.disposed.connect(() => disposables.dispose());
```

Comment:
Hello, thank you for the update. Apologies if I left behind with Theia APIs, but I remember this phosphor way of `connect` requires a `disconnect`. It's missing from this code.

```ts
const callback = () => disposable.dispose();
editorWidget.disposed.connect(callback);
disposables.push(Disposable.create(() => editorWidget.disconnect(callback)));
```

(I wrote the code on GH, it could have errors, but I hope it helps you understand my question.)

Reply:
Thanks @kittaakos, I updated the contribution. That really helps, much appreciated!

Reply:
The changes look great. I appreciate your time updating the PR multiple times